### PR TITLE
Feature/3dbump

### LINKED ISSUE #340: Add yaml configurations for BUMP_NICAS

### PR DESCRIPTION
**Pull request by guillaumevernieres**
## Description
Addition of 3 yaml files to allow using BUMP_NICAS (bump3d).

- `soca_setcorscales.yaml`: Set the parameters that define the horizontal and vertical Gaspari-Cohn function
- `soca/soca_bump3dinit.yaml`: config file for `soca_error_covariance_training.x`. Will compute the convolution weights (slow) 
- `soca/soca_3dvar_3dbump.yaml`: A bit silly, same as the 3dvar yaml but with hard-coded variables and making use of a `saber block` for **C**

### Issue(s) addressed
- fixes #340 

## Dependencies
Needs [soca-science #486](https://github.com/JCSDA-internal/soca-science/pull/486)

### CODE REVIEW

**Comment by guillaumevernieres** [comment]
No need for every body to test, it takes for ever to run and @liuxiao37k and @ShastriPaturi already volunteered!

**Review by ShastriPaturi — APPROVED** [review]
tested 1 deg on orion.

File: soca/soca_bump3dinit.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+bump:
+  verbosity: main
+  datadir: ./bump
+  method: cor
+  strategy: specific_univariate
+  new_nicas: true
+  write_nicas_local: true
+  mask_check: false #true
+  resol: 20.0
```
Comment (guillaumevernieres) [inline soca/soca_bump3dinit.yaml]: That works for the 1deg but not the 1/4 deg. I suspect this is why @liuxiao37k 's test is running out of memory.

Comment (ShastriPaturi) [reply soca/soca_bump3dinit.yaml]: @liuxiao37k are you testing the 1/4 deg on orion?

Comment (liuxiao37k) [reply soca/soca_bump3dinit.yaml]: @ShastriPaturi Yes.

**Comment by liuxiao37k** [comment]
approving it for now but in the near future we will need to 1) BGCify `soca_3dvar_3dbump.yaml` (specifically, apply bounds for `chl` under `- linear variable change name: BkgErrGODAS`. It has been making me nervous since these bounds were no longer get_or_die() but initialized as zeros if not specified in the yamls); 2) remove the hard-wired `variables` in `soca_bump3dinit.yaml` after we figure out which ones are absolutely necessary.

**Review by liuxiao37k — APPROVED** [review]

**Pull request merged**